axon-server-connector/src/main/java/org/axonframework/axonserver/connector/util/GrpcMetaDataConverter.java

```diff
@@ -74,63 +77,61 @@ public MetaDataValue convertToMetaDataValue(Object value) {
     }
 
     /**
-     * Convert the given map of MetaDataValues to a Map containing the Java representations of each MetaDataValue.
+     * Convert the given {@link Map} of {@link MetaDataValue}s to a Map containing the Java representations of each of
+     * those values.
      * <p>
      * See {@link #convertToMetaDataValue(Object)} for details about the mapping.
      *
-     * @param metaDataMap a Map containing MetaDataValue representations of each MetaData key
-     * @return a map containing the same keys, referencing to the Java representation of each corresponding value in
-     * the given {@code metaDataMap}
+     * @param metaDataMap a {@link Map} containing {@link MetaDataValue} representations of each {@link MetaData} key
+     * @return a {@link MetaData} map containing the same keys, referencing to the Java representation of each
+     * corresponding value in the given {@code metaDataMap}
      */
     public MetaData convert(Map<String, MetaDataValue> metaDataMap) {
         if (metaDataMap.isEmpty()) {
             return MetaData.emptyInstance();
         }
+
         Map<String, Object> metaData = new HashMap<>(metaDataMap.size());
         metaDataMap.forEach((k, v) -> metaData.put(k, convertFromMetaDataValue(v)));
         return MetaData.from(metaData);
     }
 
     /**
-     * Convert the given MetaDataValue to its Java representation.
+     * Convert the given {@link MetaDataValue} to its Java representation.
      * <p>
      * See {@link #convertToMetaDataValue(Object)} for details about the mapping.
      *
-     * @param value The value to convert
-     * @return The Java object representing the same value
+     * @param value the {@link MetaDataValue} to convert to its Java representation
+     * @return an {@link Object} representing the same value
      */
     public Object convertFromMetaDataValue(MetaDataValue value) {
         switch (value.getDataCase()) {
             case TEXT_VALUE:
                 return value.getTextValue();
             case BYTES_VALUE:
-                io.axoniq.axonserver.grpc.SerializedObject bytesValue = value.getBytesValue();
-                return serializer.deserialize(new SimpleSerializedObject<>(bytesValue.getData().toByteArray(),
-                                                                           byte[].class,
-                                                                           bytesValue.getType(),
-                                                                           bytesValue.getRevision()));
-
-            case DATA_NOT_SET:
-                return null;
+                return serializer.deserialize(new GrpcSerializedObject(value.getBytesValue()));
             case DOUBLE_VALUE:
                 return value.getDoubleValue();
             case NUMBER_VALUE:
                 return value.getNumberValue();
             case BOOLEAN_VALUE:
                 return value.getBooleanValue();
+            case DATA_NOT_SET:
```

Comment:
This case is the same as default...

Reply:
I shouldn't just copy paste stuff, but rather think about it. Thanks, removed the first `return null;` line.